### ISSUE TITLE
AvDateField - Custom validation

### ISSUE THREAD

**Issue by diegodesouza** [Availity/availity-react#573]
## ❓Question

Is there a way to validate the text being entered by the user? for example the user may be enter `01012020`, we would like to make sure that the format is `MM/DD/YYY`

**Comment by GoPro16** [comment]
With the newest DateField for both formik and reactstrap-validation. we already autoformat the date string from 01012020 to MM/DD/YYYY for you.

**Comment by diegodesouza** [comment]
@GoPro16 I noticed that, thanks for that feature. Except that it seems like the 'setting' of that value takes precedence, so we get `dateOfService: "Invalid date"`

**Comment by diegodesouza** [comment]
@GoPro16 we're setting the value in an `onChange`
```  
handleDateOfService = value => {
    const {store} = this.props;
    const { setDateOfService } = store;
    if (value) {
      setDateOfService(value);
    }
  };
```

```
    @action
    setDateOfService = value => {
        this.dateOfService = value;
    };

```

**Comment by GoPro16** [comment]
Ah so this seems like a bug with `@availity/reactstrap-validation-date`. In the new package we are only calling on change when the date is valid but in the old we are always calling on change.

[New World](https://github.com/Availity/availity-react/blob/master/packages/date/src/Date.js#L84-L88)
[Old World](https://github.com/Availity/availity-react/blob/master/packages/reactstrap-validation-date/src/AvDate.js#L85-L95)

We will make a change to only call on change in the old world if the date is valid.

**Comment by GoPro16** [comment]
closed via #574 

Let me know if you have any issues once the latest version has been deployed and you have installed it. The date in the on change will be in iso format.

**Comment by diegodesouza** [comment]
Perfect. Thanks @GoPro16
